lib/spack/llnl/util/filesystem.py

```diff
@@ -2354,7 +2354,10 @@ def library_dependents(self):
         """
         Set of directories where package binaries/libraries are located.
         """
-        return set([pathlib.Path(self.pkg.prefix.bin)]) | self._additional_library_dependents
+        base_pths = set()
+        if self.link_install_prefix:
+            base_pths.add(pathlib.Path(self.pkg.prefix.bin))
+        return base_pths | self._additional_library_dependents
```

Comment:
If a package implements `win_add_library_dependent`, I assume this would also likely be invalid before install time. The only example we have of that right now is in `clingo` (which refers to `python_platlib`, which is set in `python-venv` and refers to the install path of `clingo`).

Reply:
Makes sense, implemented!